# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaDecoderStateMachine.cpp
# Commit: 1d0656c6185d
# Full Hash: 1d0656c6185de116bb9cbffbcbb92319b1c07e28
# Author: alwu <alwu@mozilla.com>
# Date: 2022-11-05 21:53:50
# Regressor Bug: 1262276
# File Overlap Count: 1
# Description:
#   Bug 1262276 - part18 : add a pref to control video seamless looping. r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160869
# ==============================================================================

diff -r f436cf1a01cc -r 1d0656c6185d dom/media/MediaDecoderStateMachine.cpp
--- a/dom/media/MediaDecoderStateMachine.cpp	Sat Nov 05 02:23:35 2022 +0000
+++ b/dom/media/MediaDecoderStateMachine.cpp	Sat Nov 05 02:23:35 2022 +0000
@@ -2804,8 +2804,13 @@
                                        MediaDecoderEventVisibility::Observable);
 
   // Check whether the media satisfies the requirement of seamless looping.
-  // TODO : now we don't need to wait for metadata, can move it to other places.
+  // TODO : after we ensure video seamless looping is stable enough, then we can
+  // remove this to make the condition always true.
   mMaster->mSeamlessLoopingAllowed = StaticPrefs::media_seamless_looping();
+  if (mMaster->HasVideo()) {
+    mMaster->mSeamlessLoopingAllowed =
+        StaticPrefs::media_seamless_looping_video();
+  }
 
   SetState<DecodingFirstFrameState>();
 }